lib/segment/src/index/query_optimization/condition_converter.rs

```diff
@@ -245,6 +246,28 @@ pub fn get_float_range_checkers(
     }
 }
 
+pub fn get_int_range_checkers(
+    index: &FieldIndex,
+    range: Range<IntPayloadType>,
+) -> Option<ConditionCheckerFn> {
+    match index {
+        FieldIndex::IntIndex(num_index) => Some(Box::new(move |point_id: PointOffsetType| {
+            num_index
+                .get_values(point_id)
+                .is_some_and(|values| values.iter().copied().any(|i| range.check_range(i)))
+        })),
+        FieldIndex::FloatIndex(num_index) => {
+            let range = range.map(|i| i as FloatPayloadType);
+            Some(Box::new(move |point_id: PointOffsetType| {
+                num_index
+                    .get_values(point_id)
+                    .is_some_and(|values| values.iter().copied().any(|f| range.check_range(f)))
+            }))
+        }
+        _ => None,
+    }
+}
+
```

Comment:
For indexed fields, we follow same approach as before, converting the range into the type of the index